homeassistant/components/config/entity_registry.py

```diff
@@ -25,48 +20,40 @@
 async def async_setup(hass: HomeAssistant) -> bool:
     """Enable the Entity Registry views."""
 
-    cached_list_entities: str | None = None
-
-    @callback
-    def _async_clear_list_entities_cache(event: Event) -> None:
-        nonlocal cached_list_entities
-        cached_list_entities = None
-
-    @websocket_api.websocket_command(
-        {vol.Required("type"): "config/entity_registry/list"}
-    )
-    @callback
-    def websocket_list_entities(
-        hass: HomeAssistant,
-        connection: websocket_api.ActiveConnection,
-        msg: dict[str, Any],
-    ) -> None:
-        """Handle list registry entries command."""
-        nonlocal cached_list_entities
-        if not cached_list_entities:
-            registry = er.async_get(hass)
-            cached_list_entities = message_to_json(
-                websocket_api.result_message(
-                    IDEN_TEMPLATE,  # type: ignore[arg-type]
-                    [_entry_dict(entry) for entry in registry.entities.values()],
-                )
-            )
-        connection.send_message(
-            cached_list_entities.replace(IDEN_JSON_TEMPLATE, str(msg["id"]), 1)
-        )
-
-    hass.bus.async_listen(
-        er.EVENT_ENTITY_REGISTRY_UPDATED,
-        _async_clear_list_entities_cache,
-        run_immediately=True,
-    )
     websocket_api.async_register_command(hass, websocket_list_entities)
     websocket_api.async_register_command(hass, websocket_get_entity)
     websocket_api.async_register_command(hass, websocket_update_entity)
     websocket_api.async_register_command(hass, websocket_remove_entity)
     return True
 
 
+@websocket_api.websocket_command({vol.Required("type"): "config/entity_registry/list"})
+@callback
+def websocket_list_entities(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Handle list registry entries command."""
+    registry = er.async_get(hass)
+    # Concatenate cached entity registry item JSON serializations
+    entries_json = (
+        "["
+        + ",".join(
+            entry.json_repr
+            for entry in registry.entities.values()
+            if entry.json_repr is not None
+        )
+        + "]"
+    )
+    # Build response message
+    msg_json = (
+        f'{{"id":{msg["id"]},"type": "{websocket_api.const.TYPE_RESULT}",'
+        f'"success":true,"result": {entries_json}}}'
```

Comment:
I think it would be cheaper to inline the string build here.  It looks like its going to build the large string above and than have to build another one here.